json_schema_fuzz/__init__.py

```diff
@@ -1,15 +1,101 @@
 """JSON schema fuzzer."""
+import json
 import random
 import string
+from decimal import Decimal
 
 import exrex
 
 from .schema_operations import merge
 
+MAX_REJECTED_SAMPLES = 1000
+
+
+class RejectionSamplingFailed(Exception):
+    """
+    Failed to generate sample that satisfies all criteria
+    """
+
+
+def listify(value):
+    """ If value is not a list wrap it in a list """
+    if isinstance(value, list):
+        return value
+    else:
+        return [value]
+
 
 def random_integer(schema):
     """Generate random integer."""
-    return random.randint(0, 10)
+
+    multiple_of = schema.get("multipleOf", Decimal(1))
+
+    default_min = Decimal(1 * multiple_of)
+    default_max = Decimal(100 * multiple_of)
+
+    minimum = max(
+        schema.get("minimum", default_min),
+        schema.get("exclusiveMinimum", default_min) + 1
+    )
+    maximum = min(
+        schema.get("maximum", default_max),
+        schema.get("exclusiveMaximum", default_max) - 1
+    )
+
+    not_multiple_of = schema.get("notMultipleOf", [])
+    if not isinstance(not_multiple_of, list):
+        not_multiple_of = [not_multiple_of]
+
+    for _ in range(MAX_REJECTED_SAMPLES):
+        # Generate new value
+        value = random.randint(minimum, maximum)
+
+        # Verify
+        if value % multiple_of != 0:
+            continue
+        is_multiple_of = [value % num == 0 for num in not_multiple_of]
+        if any(is_multiple_of):
+            continue
+        return int(value)
+    raise RejectionSamplingFailed()
+
+
+def random_number(schema):
+    """Generate random number."""
+    default_min = Decimal(1)
+    default_max = Decimal(100)
+    decimal_digits = 4
+
+    smallest_interval = Decimal(10 ** -decimal_digits)
+
+    minimum = max(
+        schema.get("minimum", default_min),
+        schema.get("exclusiveMinimum", default_min) - smallest_interval
```

Comment:
I think you want `+` here, but I also think it's unnecessary. When uniformly, continuously sampling, your chances of getting the endpoint exactly are infinitesimal.

Reply:
That would be true if we were using floats but I don't think that is true here. We are using decimals, so it is very possible that we hit the value of an exclusive minimum.